app/views/posts/_posts.html.erb

```diff
@@ -0,0 +1,45 @@
+<% posts.each_with_index do |post, index| %>
+    <div class="card mt-3">
+
+      <%= link_to user_post_path(post.author, post), class: "text-decoration-none text-secondary" do %>
+        <div class="card-header d-flex justify-content-between">
+          <span class="text-success">
+            <strong>
+              Post #<%= index + 1 %>
+              <%= posts.size == 1 ? " | By #{post.author.name}" : "" %>
+            </strong>
+          </span>
+          <% if posts.size == 1 %> 
+            <div class="text-muted">
+              <span>Comments: <%= post.comments_counter %></span>,
+              <span>Likes: <%= post.likes_counter %></span>
+            </div>
+          <% end %>
+        </div>
+
+        <div class="card-body">
+          <h5 class="card-title"><%= post.title %></h5>
+          <p class="card-text"><%= truncate_n.nil? ? post.text : post.text.truncate(truncate_n) %></p>
+        </div>
+
+        <% if posts.size > 1 %>
+          <div class="text-muted pt-1 pb-2 pr-3" style="text-align: end;">
+            <span>Comments: <%= post.comments_counter %></span>,
+            <span>Likes: <%= post.likes_counter %></span>
+          </div>
+        <% end %>
+      <% end %>
+    </div>
+
+    <% if limit_n > 0 %>
+      <div class="card card-body mt-1 p-3">
+        <ul class="list-unstyled">
+
+          <% post.most_recent_comments(limit_n).each do |comment| %>
+            <li><%= comment.author.name %>: <%= comment.text %></li>
+          <% end %>
+
+        </ul>
+      </div>
+    <% end %>
```

Comment:
- [ ] I could not confirm the comment because I ran into an error while trying to add data, others were added successfully but the comment failed to, I will encourage you to double-check it.
![image](https://github.com/ITurres/blog_app/assets/86200356/d8b2d022-a8b4-45b2-a30e-823515b3da20)

Reply:
Hi @Olamarx! it is running well on my end! 🎯

Reply:
It runs well on my end too, just the comments were not added.

Reply:
@Olamarx, a short screencast
[addind-data-to-blog-db-dev.webm](https://github.com/ITurres/blog_app/assets/100724715/5457f5a4-57c0-4c7b-a95c-73cb495b1441)

Reply:
@Olamarx, it is because when instantiating `Comment` is expecting `author` as a param, in your instantiations you send `user`.

Reply:
i.e

```rb
   Comment.create(post: first_post, author: first_user, text: 'Hi Tom!' )
```